### PR TITLE
Feature 3522 lbds prototype final

### PR DESCRIPTION
**Pull request by syntakker**
Fixes 3522#

### CODE REVIEW

**Comment by syntakker** [comment]
To do:
- locally encrypt LBDS private key and AES secret
- resolve merge conflicts

File: sormas-app/app/src/main/java/de/symeda/sormas/app/settings/SettingsActivity.java [hunk 1]
```diff
@@ -82,6 +82,8 @@ public boolean onOptionsItemSelected(MenuItem item) {
 			String serverUrl = getActiveFragment().getServerUrl();
 			ConfigProvider.setServerRestUrl(serverUrl);
 
+			getActiveFragment().saveLbdsDebugUrl();
```
Comment (MartinWahnschaffe) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/settings/SettingsActivity.java:85]: should be done the same way as we are doing it for the server rest url, to have it consistent:

```
String lbdsDebugUrl= getActiveFragment().getLbdsDebugUrl();
ConfigProvider.setServerLbdsDebugUrl(lbdsDebugUrl);
```

File: sormas-app/app/src/main/java/de/symeda/sormas/app/settings/SettingsFragment.java [hunk 2]
```diff
@@ -273,6 +293,31 @@ public void logout() {
...
+	}
+
+	public void saveLbdsDebugUrl() {
+		if (isLbdsAppInstalled() && isShowDevOptions()) {
+			String debugUrl = binding.settingsLbdsDebugUrl.getValue();
+			ConfigProvider.setServerLbdsDebugUrl(debugUrl);
+		}
+	}
+
+	private void resetFields(PersonDto personDto) {
```
Comment (MartinWahnschaffe) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/settings/SettingsFragment.java:317]: Remove?!

File: sormas-app/app/src/main/res/values/strings_format.xml [hunk 3]
```diff
@@ -23,6 +23,10 @@
     <string name="age_with_dob_and_prepend_format"><b>%1$s:</b> %2$s %3$s (%4$s)</string>
     <string name="date_range_format">%1$s - %2$s</string>
     <string name="due_timeago_format">due %1$s</string>
+    <string name="lbds_response_cases">LBDS Sync Response. Cases transferred: %1$d , ignored: %2$d</string>
+    <string name="lbds_response_persons">LBDS Sync Response. Persons transferred: %1$d , ignored: %2$d</string>
+    <string name="lbds_cases_transferred">LBDS Sync. Cases transferred: %1$d</string>
```
Comment (MartinWahnschaffe) [inline sormas-app/app/src/main/res/values/strings_format.xml:28]: Better:
<string name="lbds_send_cases">LBDS Sync. Sending cases: %1$d</string>

"transferred" sound like it's already done

**Review by MartinWahnschaffe — CHANGES_REQUESTED** [review]
@syntakker Just a few small things. Afterwards you can merge the PR

File: sormas-app/app/src/main/java/de/symeda/sormas/app/LbdsRecevierComponent.java [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+import retrofit2.Response;
+
+public class LbdsRecevierComponent extends IntentService {
+
+	/**
+	 * Creates an IntentService. Invoked by your subclass's constructor.
+	 */
+	public LbdsRecevierComponent() {
+		super("LbdsRecevierComponent");
+		// super(SormasRelated.SORMAS_RECEIVER_COMPONENT);
```
[the remark this answers is not in the document]
Comment (syntakker) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/LbdsRecevierComponent.java:43]: done.

File: sormas-app/app/src/main/java/de/symeda/sormas/app/backend/config/ConfigProvider.java [hunk 5]
```diff
@@ -761,4 +781,147 @@ public static void setInitialSyncRequired(boolean initialSyncRequired) {
...
+	}
+
+	public static PrivateKey getLbdsSormasPrivateKey() {
+		if (instance.lbdsSormasPrivateKey == null) {
+			Config config = DatabaseHelper.getConfigDao().queryForId(LBDS_SORMAS_PRIVATE_KEY);
+			if (config != null) {
+				try {
+					String aesKey = getLbdsSormasPrivateKeyAesSecret();
+					String lbdsSormasPrivateKeyPKCS8 = LbdsKeyHelper.decryptAES(config.getValue(), aesKey);
+					instance.lbdsSormasPrivateKey = LbdsKeyHelper.getPrivateKeyFromPKCS8(lbdsSormasPrivateKeyPKCS8);
```
[the remark this answers is not in the document]
Comment (syntakker) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/backend/config/ConfigProvider.java:822]: done.

File: sormas-app/app/src/main/java/de/symeda/sormas/app/settings/SettingsActivity.java (hunk 1 above)
Comment (syntakker) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/settings/SettingsActivity.java:85]: done

File: sormas-app/app/src/main/java/de/symeda/sormas/app/settings/SettingsFragment.java (hunk 2 above)
Comment (syntakker) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/settings/SettingsFragment.java:317]: done

File: sormas-app/app/src/main/res/values/strings_format.xml (hunk 3 above)
Comment (syntakker) [reply sormas-app/app/src/main/res/values/strings_format.xml:28]: done

**Pull request merged**